x-pack/plugins/cloud_security_posture/common/utils/detection_rules.test.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CspBenchmarkRuleMetadata } from '../types';
+import {
+  convertRuleTagsToKQL,
+  generateBenchmarkRuleTags,
+  getFindingsDetectionRuleSearchTags,
+} from './detection_rules';
+
+describe('Tests detection rules utils', () => {
```

Comment:
all the tests here are fine but they only test the green state, i would also like to see a test for a red state. for example:
```
it('handling undefined in tags input', () => {
    const inputTags = ['tag1', undefined, 'tag3'];

    const result = convertRuleTagsToKQL(inputTags);

    const expectedKQL = 'alert.attributes.tags:("tag1" AND "tag3")';
    expect(result).toBe(expectedKQL);
  });
```

This is important because we don't have an unexpected result like: `"tag1" AND undefined AND "tag3"`

This same example can also be implemented for all other tests that you did here